### ISSUE TITLE
Regularized incomplete beta and gamma at half integers

### ISSUE THREAD

**Issue by Beakerboy** [markrogoyski/math-php#127]
I'm thinking about splitting out the "exact" calculations of these functions into their own methods. I was thinking it would probably be best for the base cases to be continuous, and not have a different calculation at the half integers. If another method is gauranteed to pass on half-integer values, we can make them call the gammaExact or regularizedIncompleteBetaExact method.

As far as type hinting goes, I'm thinking of trying to avoid floats, and having to test that they are close to a half integer. If I is a function of a and b, should I ask for a*2 to type hint it as an int, or would that be confusing. Alternatively, I could ask for an int numerator and int denominator of each, where the denominator is either 1 or 2.

**Comment by markrogoyski** [comment]
From the user perspective, they should not have to know about the implementation. They just know they want to call the gamma function and have it return a correct result. It is fine if there are variations of a function for specific purposes, as long as the general functions work for everything in a reasonable manner and the user can use those with confidence without needing to know about the special versions of the functions.

For example, the user code has some calculation that they then run the gamma function on. They are not going to want to have checks for integers and conditional logic to call specific functions based on our implementation.

Basically, the generic gamma, regularizedIncompleteBeta etc. functions should work for all input. Then, having specific versions for special use cases is fine. This sounds like what you are thinking so no problem.
